api/alerts.go

```diff
@@ -131,6 +130,12 @@ func (a *Alerts) Add(viewName string, newAlert *Alert) (*Alert, error) {
 		field := graphql.String(newAlert.ThrottleField)
 		throttleField = &field
 	}
+	var runAsUserIdPtr *graphql.String
+	if runAsUserID != "" {
+		runAsUserIdPtr = graphql.NewString(graphql.String(runAsUserID))
+	} else {
+		runAsUserIdPtr = (*graphql.String)(nil)
```

Comment:
runAsUserId cannot be "" when used as input, so we need a nil pointer

Reply:
an alternative is to create different mutation structs, but that becomes quite messy if we need to add more optional variables